### ISSUE TITLE
Build failed on Cygwin when C++03 specified

### ISSUE THREAD

**Issue by mmidor** [cpputest/cpputest#654]
Since the target DSP compiler only supports C++03, I am forced to build CppUTest using mentioned standard. The compilation fails on \Gcc\UtestPlatform.cpp, `PlatformSpecificVSNprintf()`. The functions `snprintf/vsnprintf` are not supported in this standard.

**Comment by basvodde** [comment]
Oh, they aren't? How would they implement it then?

**Comment by mmidor** [comment]
Yes, the **safe** version of mentioned functions are only available when C99 or C++11 is specified.
Because this function needs special treatment on many platforms, maybe it can be implemented independently? In my project I am using small `snrprintf()` from AVR library, with a few modifications, supporting I think 90% of full `sprintf` library features. What do you think about this?

**Comment by basvodde** [comment]
On the long run, the vsnprintf functionality is on the list of implementing ourselves... so that we have less platform dependencies. On the short run, I suggest to fix it with a #ifdef in the PlatformSpecific file (checking on C version) and call the non-secure versions?

**Comment by arstrube** [comment]
@mmidor it's been a while.... 

I presume ` snrprintf()` is available in source? Are any copyright restrictions attached to it? We copied a lot of the C string functions that are now in SimpleString from various sources, because most of these are very well-known algorithms, where copyright isn't really an issue, because it has been done like that for decades... 

If the copyright issue can be sorted out, then I would think `snrprintf()` might be a good starting point for our own implementation of this functionality and taking it out of the platform-specific file entirely.

What do you think @basvodde ?

**Comment by basvodde** [comment]
I'd prefer to write it and test-drive it than to copy an existing version. We can use an existing version as guide, but not just copy it.

**Comment by arstrube** [comment]
That's basically more or less what I meant. Test-drive-write-along-the-lines-of-the-existing-code-a-piece-at-a-time ;-). I suppose since that essentially means a re-write, then copyright would be a non-issue?

I think this type of stuff is fairly involved and having a good 'prototype' might make the difference.
